### DISCUSSION TITLE
Binding Grid ColumnDefinitions for dynamic column count

### DISCUSSION THREAD

**Discussion by moon6969** [AvaloniaUI/Avalonia#12911] [Q&A] [answered]
I'm trying to use a SharedSizeScope Grid as an ItemsControl.ItemsPanel, so I want to bind the Grid's column count to the ItemsSource count.

Is there a cool Avalonia way to do this?
If not, I will try some attached properties à la [Rachel's WPF Grid’s Row/Column Count Properties](https://rachel53461.wordpress.com/2011/09/17/wpf-grids-rowcolumn-count-properties/).

NB: I can't do it via the "list format" since I need to also set SharedSizeGroup... 
`<Grid ColumnDefinitions="{Binding ColumnDefs}">`
I did try the above, but was getting _Unable to find suitable setter or adder_ error - should it work?

Thanks.

**Comment by timunie** [comment, marked as answer]
That's an attached property in the sample and an attached property would work for Avalonia as well. If all columns have the same width, you can also use `UniformGrid` which is way simpler. 

https://docs.avaloniaui.net/docs/next/concepts/attached-property
